src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook_test.go

```diff
@@ -633,6 +633,18 @@ func TestValidateUpdate_NoError(t *testing.T) {
 		assert.Error(t, err)
 	})
 
+	t.Run("cluster can be deleted even if licenseRef not found", func(t *testing.T) {
```

Comment:
Am I right in thinking that when we create cluster that has valid licenseRef. Then we remove the secret with the license we would not be able to update cluster?

Reply:
Yes since it is part of the spec. If user wants to delete the license, it should also be removed in the spec - not only the referenced secret.

Reply:
My personal preference would be to flag that there is a problem with the license during updates. Any update should be possible once cluster have some license.

Reply:
The desired state, spec, wouldn't be correct then as the licenseRef in the spec is ignored once the license is loaded (e.g. the referenced secret can be deleted once it is loaded?)

Reply:
Maybe it is debatable if license kubernetes secret is gone, then this should prevent any update to the cluster custom resource.

My preference would be to log an error rather than fail it hard. Other solution would be to disable whole validation (disable webhook) in case of an incident.

Reply:
Or just delete the licenseRef as well in case of incident, the license is not enforced at Redpanda